File: Backend/C#/Backend/Models/Item.cs
using System.ComponentModel.DataAnnotations;
using Backend.Validators;

namespace Backend.Models {
	public class Item {
		
		public int Id { get; set; }

		[Required]
		[MaxLength(127)]
		public string Name { get; set; }

		[Required]
		[MaxLength(4095)]
		public string Description { get; set; }

		[Required]
		[MaxLength(511)]
		[Url]
		public string Url { get; set; }
		
		[Required]
		public bool Available { get; set; }
		
		[Required]
		[MinValue(0)]
		public int Price { get; set; }

		public Item() {}

		public Item(int price, string name, string description, string url, bool available, int id = 0) {
			Id = id;
			Price = price;
			Name = name;
			Description = description;
			Url = url;
			Available = available;
		}

		public override string ToString() {
			return $"{nameof(Id)}: {Id}, {nameof(Price)}: {Price}, {nameof(Name)}: {Name}, {nameof(Description)}" +
			       $": {Description}, {nameof(Url)}: {Url}, {nameof(Available)}: {Available}";
		}
	}
}

File: Backend/java/SpringBoot/src/main/java/nl/bookshop/models/Item.java
package nl.bookshop.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;
import org.hibernate.validator.constraints.URL;

import javax.persistence.*;
import javax.validation.constraints.*;

import static nl.bookshop.models.ModelStrings.*;
import static nl.bookshop.validators.ValidationStrings.NEGATIVE_PRICE;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = ITEMS)
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    @Min(value = 0, message = NEGATIVE_PRICE)
    private Integer price;

    @NotNull
    @Length(max = 127)
    private String name;

    @NotNull
    @Length(max = 4095)
    private String description;

    @NotEmpty
    @URL
    @Length(max = 511)
    private String url;

    @NotNull
    private boolean available;
}
